src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -166,13 +166,10 @@ void GcsResourceManager::HandleReportResourceUsage(
   auto resources_data = std::make_shared<rpc::ResourcesData>();
   resources_data->CopyFrom(request.resources());
 
-  // We use `node_resource_usages_` to filter out the nodes that report resource
-  // information for the first time. `UpdateNodeResourceUsage` will modify
-  // `node_resource_usages_`, so we need to do it before `UpdateNodeResourceUsage`.
-  if (node_resource_usages_.count(node_id) == 0 ||
-      resources_data->resources_available_changed()) {
-    const auto &resource_changed = MapFromProtobuf(resources_data->resources_available());
-    SetAvailableResources(node_id, ResourceSet(resource_changed));
+  const auto &resource_changes =
+      MapFromProtobuf(resources_data->normal_task_resources_changes());
+  if (!resource_changes.empty()) {
+    UpdateNormalTaskResourcesChanges(node_id, resource_changes);
```

Comment:
I think we don't have centralized view for actors now? I think eventually we need some RAY_CHECK to make sure the addition of task usage + actor usage are equal to the total resources.

Reply:
Yeah, we need some RAY_CHECK to make sure the addition of task usage + actor usage are equal to the total resources. Actor creation has not used `gcs_resource_manager` yet, so i will implement it in the next pr.

Reply:
Hey @wuisawesome @rkooo567 If we enable new scheduling, the resource used by task may exceed that of node resource? I found that a node with CPU 1 can schedule two normal tasks (each using 1 CPU).

Reply:
Ah yeah there are times that we borrow cpus to other workers (e.g., when we call ray.get or other remote methods inside ray task or actor to avoid deadlock). It is probably that one.